packages/pcb-stackup-core/example/index.js

```diff
@@ -47,22 +48,28 @@ function renderStackupAndWrite (layers, done) {
 }
 
 function renderAllLayers (done) {
-  const tasks = GERBER_PATHS.map(file => next => renderLayer(file, next))
+  const layerTypes = wtg(GERBER_PATHS)
+  const tasks = GERBER_PATHS.map(file => next =>
+    renderLayer(file, layerTypes, next)
+  )
 
   runParallel(tasks, done)
 }
 
-function renderLayer (filename, done) {
+function renderLayer (filename, layerTypes, done) {
   const file = fs.createReadStream(filename)
-  const type = whatsThatGerber(filename)
+  const layer = layerTypes[filename]
+
+  assert(layer, `Expected ${filename} to be recognized as a gerber`)
```

Comment:
This assert needs to be moved down to assert on `type`, or maybe just removed. It was guarding against null return, which was removed in favor of nulling the type and side values to have a consistent return shape